openmetadata-ui/src/main/resources/ui/src/components/common/table-data-card-v2/TableDataCardV2.tsx

```diff
@@ -0,0 +1,193 @@
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+import { faExclamationCircle } from '@fortawesome/free-solid-svg-icons';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { isNil, isString, startCase, uniqueId } from 'lodash';
+import { ExtraInfo } from 'Models';
+import React, { useMemo } from 'react';
+import { Link, useLocation } from 'react-router-dom';
+import AppState from '../../../AppState';
+import { FQN_SEPARATOR_CHAR } from '../../../constants/char.constants';
+import { ROUTES } from '../../../constants/constants';
+import { EntityType, FqnPart } from '../../../enums/entity.enum';
+import { SearchIndex } from '../../../enums/search.enum';
+import { CurrentTourPageType } from '../../../enums/tour.enum';
+import { OwnerType } from '../../../enums/user.enum';
+import { EntityReference } from '../../../generated/entity/type';
+import {
+  getEntityId,
+  getEntityName,
+  getEntityPlaceHolder,
+  getOwnerValue,
+  getPartialNameFromTableFQN,
+} from '../../../utils/CommonUtils';
+import { serviceTypeLogo } from '../../../utils/ServiceUtils';
+import { stringToHTML } from '../../../utils/StringsUtils';
+import { getEntityLink } from '../../../utils/TableUtils';
+import { SearchedDataProps } from '../../searched-data/SearchedData.interface';
+import '../table-data-card/TableDataCard.style.css';
+import TableDataCardBody from '../table-data-card/TableDataCardBody';
+
+export interface TableDataCardPropsV2 {
+  id: string;
+  source: SearchedDataProps['data'][number]['_source'];
+  matches?: {
+    key: string;
+    value: number;
+  }[];
+  searchIndex: SearchIndex | EntityType;
+}
+
+const TableDataCardV2: React.FC<TableDataCardPropsV2> = ({
+  id,
+  source,
+  matches,
+  searchIndex,
+}) => {
+  const location = useLocation();
+
+  const otherDetails: ExtraInfo[] = [];
+
+  if ('tier' in source && !isNil(source.tier)) {
+    otherDetails.push({
+      key: 'Tier',
+      value: isString(source.tier)
+        ? source.tier
+        : source.tier.tagFQN.split(FQN_SEPARATOR_CHAR)[1],
+    });
+  }
+
+  if ('owner' in source && !isNil(source.owner)) {
+    otherDetails.push({
+      key: 'Owner',
+      value: getOwnerValue(source.owner as EntityReference),
+      placeholderText: getEntityPlaceHolder(
+        getEntityName(source.owner as EntityReference),
+        source.owner?.deleted
+      ),
+      id: getEntityId(source.owner as EntityReference),
+      isEntityDetails: true,
+      isLink: true,
+      openInNewTab: false,
+      profileName:
+        source.owner?.type === OwnerType.USER ? source.owner?.name : undefined,
+    });
+  }
+
+  if ('usageSummary' in source) {
+    otherDetails.push({
+      key: 'Usage',
```

Comment:
Can we wrap this to useMemo with dependancy of `source` and other required props?